schema/field/field.go

```diff
@@ -298,6 +298,27 @@ func (b *stringBuilder) Descriptor() *Descriptor {
 	return b.desc
 }
 
+// customTypeStringBuilder is the builder for string fields that have overridden the type via GoType.
+// This wraps stringBuilder but relaxes the signature of DefaultFunc.
+type customTypeStringBuilder struct {
+	*stringBuilder
+}
+
+// DefaultFunc sets the function that is applied to set the default value
+// of the field on creation. For example:
+//
+//	field.String("cuid").
+//		DefaultFunc(cuid.New)
+//
+func (b *customTypeStringBuilder) DefaultFunc(fn interface{}) *customTypeStringBuilder {
+	typ := reflect.TypeOf(fn)
+	if typ.Kind() != reflect.Func || typ.NumIn() != 0 || typ.NumOut() != 1 || typ.Out(0).String() != b.desc.Info.String() {
+		b.desc.Err = fmt.Errorf("expect type (func() %s) for default value", b.desc.Info)
+	}
```

Comment:
What do you think about moving this check to the `Descriptor` method, and changing the function signature of the `stringBuilder`? I think we need to run this check even if the `GoType` option is called after the default function.

Reply:
That sounds cleaner to me, let me prep that.

Reply:
As I'm adding this it seems as though this should exist on the other builder types as well..